libs/beacon/src/beacon_service.cpp

```diff
@@ -487,37 +500,43 @@ BeaconService::State BeaconService::OnVerifySignaturesState()
   }
 
   // Don't lock until the promise has resolved! Otherwise the system can deadlock.
-  std::lock_guard<std::mutex> lock(mutex_);
-
-  if (ret.threshold_signatures.empty())
   {
-    FETCH_LOG_INFO(LOGGING_NAME, "Peer wasn't ready when asking for signatures: ",
-                   qual_promise_identity_.identifier().ToBase64());
-    state_machine_->Delay(std::chrono::milliseconds(10));
-    return State::COLLECT_SIGNATURES;
-  }
+    std::lock_guard<std::mutex> lock(mutex_);
 
-  if (ret.round != current_entropy_.round)
-  {
-    FETCH_LOG_WARN(LOGGING_NAME, "Peer returned the wrong round when asked for signatures. Peer: ",
-                   qual_promise_identity_.identifier().ToBase64(), " returned: ", ret.round,
-                   " expected: ", current_entropy_.round);
-    return State::COLLECT_SIGNATURES;
-  }
+    if (ret.threshold_signatures.empty())
+    {
+      FETCH_LOG_INFO(LOGGING_NAME, "Peer wasn't ready when asking for signatures: ",
+                     qual_promise_identity_.identifier().ToBase64());
+      state_machine_->Delay(std::chrono::milliseconds(100));
 
-  // Success - Add relevant info
-  auto &signatures_struct = signatures_being_built_[current_entropy_.round];
-  auto &all_sigs_map      = signatures_struct.threshold_signatures;
+      return State::COLLECT_SIGNATURES;
+    }
 
-  for (auto const &address_sig_pair : ret.threshold_signatures)
-  {
-    all_sigs_map[address_sig_pair.first] = address_sig_pair.second;
-    // Let the manager know
-    AddSignature(address_sig_pair.second);
-  }
+    if (ret.round != current_entropy_.round)
+    {
+      FETCH_LOG_WARN(LOGGING_NAME,
+                     "Peer returned the wrong round when asked for signatures. Peer: ",
+                     qual_promise_identity_.identifier().ToBase64(), " returned: ", ret.round,
+                     " expected: ", current_entropy_.round);
+      return State::COLLECT_SIGNATURES;
+    }
+
+    // Success - Add relevant info
+    auto &signatures_struct = signatures_being_built_[current_entropy_.round];
+    auto &all_sigs_map      = signatures_struct.threshold_signatures;
+
+    for (auto const &address_sig_pair : ret.threshold_signatures)
+    {
+      all_sigs_map[address_sig_pair.first] = address_sig_pair.second;
+      // Let the manager know
+      AddSignature(address_sig_pair.second);
+    }
 
-  FETCH_LOG_INFO(LOGGING_NAME, "After adding, we have ", all_sigs_map.size(),
-                 " signatures. Round: ", current_entropy_.round);
+    FETCH_LOG_INFO(LOGGING_NAME, "After adding, we have ", all_sigs_map.size(),
+                   " signatures. Round: ", current_entropy_.round);
+  }  // Mutex unlocks here since verification can take some time
+
+  MilliTimer const timer{"Verify threshold signature", 100};
```

Comment:
this prints if it takes longer than 100 ms from construction to destruction of the object